packages/grafana-data/src/transformations/transformers/joinByField.test.ts

```diff
@@ -1011,21 +1056,26 @@ describe('JOIN Transformer', () => {
 
     it('handles duplicate field name', async () => {
       const cfg: DataTransformerConfig<JoinByFieldOptions> = {
-        id: DataTransformerID.seriesToColumns,
+        id: DataTransformerID.joinByField,
         options: {
-          byField: 'time',
+          fields: {
+            frame1: 'time',
+            frame2: 'time',
+          },
           mode: JoinMode.inner,
         },
       };
 
       const frame1 = toDataFrame({
+        refId: 'frame1',
         fields: [
           { name: 'time', type: FieldType.time, values: [1] },
           { name: 'temperature', type: FieldType.number, values: [10] },
         ],
       });
 
       const frame2 = toDataFrame({
+        refId: 'frame2',
```

Comment:
Great job on this! Would it be possible to include a gdev dashboard for this transform? We can deploy / share gdev dashboards via play.grafana.org. We can provide a link to the public dashboard in release notes so users can try out changes easily. They can also help reviewers of the PR test out the changes

Reply:
see [this doc](https://docs.google.com/document/d/1Zt_MHSWCd9vNX84vdVcHi4y-WuoZB86tB8ezZl_i3Qs/edit?usp=sharing) I wrote up explaining more on gdev dashboards 👍

Reply:
Added a dashboard

![Screenshot_20230118_011125](https://user-images.githubusercontent.com/9373203/213168286-866d9984-a19f-4719-b9d8-1578b72919f9.png)